### PR TITLE
SMS authentication support

### PR DESCRIPTION
**Pull request by ivancevich**

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -1430,6 +1433,65 @@ Auth0.prototype.getConnections = function (callback) {
 };
 
 /**
+ * Send SMS to do passwordless authentication
+ *
+ * Examples:
```
Comment (pose) [inline lib/index.js]: You can use @example here, thanks!

File: lib/index.js [hunk 2]
```diff
@@ -1430,6 +1433,65 @@ Auth0.prototype.getConnections = function (callback) {
...
+      Authorization: 'Bearer ' + apiToken
+    },
+    data:         {
+      phone_number:   phoneNumber,
+      connection:     'sms',
+      email_verified: false
+    }
+  })
+  .fail(function (err) {
+    try {
```
Comment (pose) [inline lib/index.js]: There are a couple of failure corner cases handled by this function: https://github.com/auth0/auth0.js/blob/master/lib/index.js#L84-L120 . Do you think we can re use that function here?

Comment (pose) [inline lib/index.js]: Also, there's a flag `this._useJSONP` that we use in other methods: https://github.com/auth0/auth0.js/blob/master/lib/index.js#L303

**Comment by pose** [comment]
Apart from the comments, looks good to me :+1:

File: lib/index.js (hunk 2 above)
Comment (cristiandouce) [inline lib/index.js]: @pose You cant use jsonp with APIv2. :dancers:

Comment (pose) [inline lib/index.js]: Sorry, @cristiandouce is right :). Disregard this comment :dancer:

**Pull request merged**